### PR TITLE
Add modal with list of related entities 

### PR DESCRIPTION
**Pull request by mournfulCoroner**

### CODE REVIEW

**Comment by mournfulCoroner** [comment]
Related to https://github.com/datalens-tech/datalens/issues/81

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/ru/">component.dialog-related-entities.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/en/">component.dialog-related-entities.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/ru/">component.dialog-related-entities.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/en/">component.dialog-related-entities.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/ru/">component.dialog-related-entities.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/en/">component.dialog-related-entities.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/ru/">component.dialog-related-entities.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-dialog-related-entities-view-1133/en/">component.dialog-related-entities.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/ru/">component.entry-context-menu.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-context-menu-view-1133/en/">component.entry-context-menu.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/ru/">component.entry-row.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9701-related-entities-list-1133/component-entry-row-view-1133/en/">component.entry-row.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

File: src/ui/components/DialogRelatedEntities/DialogRelatedEntities.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        if (
+            isEmpty(relations) &&
+            entry.scope === EntryScope.Widget &&
+            Object.values(EDITOR_TYPE).includes(entry.type) &&
+            currentDirection === Direction.Parent
+        ) {
+            return <Alert theme="warning" message={i18n('label_editor-hint')} />;
+        }
+
+        if (!relations || isEmpty(relations) || isError) {
```
Comment (flops) [inline src/ui/components/DialogRelatedEntities/DialogRelatedEntities.tsx:103]: I think better to separate this if like this and move this logic right after isLoading check

```
if (isError) {
 return (
    <div className={b('error-state')}>
      <PlaceholderIllustration direction="column" name={'error'} title={i18n('label_request-error')} />
    </div>
 )
}
```

And `!relations  || isEmpty(relations)` when relations is [] | null is `isEmpty(relations)` (`isEmpty(null) -> true`)

In that case maybe:

```
if (isEmpty(relations)) {
   if (entry.scope === EntryScope.Widget &&
            Object.values(EDITOR_TYPE).includes(entry.type) &&
            currentDirection === Direction.Parent) { return <Alert/> }
  

  return (<div className={b('empty-state')}/>)
}
```

File: src/ui/components/DialogRelatedEntities/EntitiesList/EntitiesList.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        case EntryScope.Dash:
+            return i18n('label_scope-dash');
+        case EntryScope.Connection:
+            return i18n('label_scope-connection');
+        case EntryScope.Dataset:
+            return i18n('label_scope-dataset');
+        case EntryScope.Widget:
+            return i18n('label_scope-chart');
+        default:
+            return i18n('label_scope-chart');
```
Comment (flops) [inline src/ui/components/DialogRelatedEntities/EntitiesList/EntitiesList.tsx:32]: `label_scope-other`

**Review by flops — CHANGES_REQUESTED** [review]

**Review by flops — APPROVED** [review]

**Pull request merged**